libs/product/src/selectors/composite-product/composite-product.selectors.ts

```diff
@@ -29,6 +36,108 @@ const createCompositeProductSelectors = (): DaffCompositeProductMemoizedSelector
 		selectProductDiscountAmount
 	} = getDaffProductEntitiesSelectors();
 
+	/**
+	 * Selector for the minimum price possible for a composite product, regardless of the current item option selection.
+	 * This could be useful for a quick preview of the product.
+	 */
+	const selectCompositeProductMinPossiblePrice = createSelector(
+		selectProductEntities,
+		(products, props) => {
+			const product = selectProduct.projector(products, { id: props.id });
+			if(product.type !== DaffProductTypeEnum.Composite) {
+				return undefined;
+			}
+			
+			return (<DaffCompositeProduct>product).items.reduce((acc, item) => 
+				daffAdd(
+					acc, 
+					getMinimumCompositeItemPrice(item)
+				), product.price);
+		}
+	);
+
+	/**
+	 * Selector for the maximum price possible for a composite product, regardless of the current item option selection.
+	 * This could be useful for a quick preview of the product.
+	 */
+	const selectCompositeProductMaxPossiblePrice = createSelector(
+		selectProductEntities,
+		(products, props) => {
+			const product = selectProduct.projector(products, { id: props.id });
+			if(product.type !== DaffProductTypeEnum.Composite) {
+				return undefined;
+			}
+			
+			return (<DaffCompositeProduct>product).items.reduce((acc, item) => 
+				daffAdd(
+					acc, 
+					Math.max(...item.options.map(option => option.price))
+				), product.price);
+		}
+	);
+
+	/**
+	 * Selector for whether the composite product could have a price range.
+	 */
+	const selectCompositeProductPossiblyHasPriceRange = createSelector(
+		selectProductEntities,
+		(products, props) => selectCompositeProductMinPossiblePrice.projector(products, { id: props.id }) 
+			!== selectCompositeProductMaxPossiblePrice.projector(products, { id: props.id })
+	)
+
+	/**
+	 * Selector for the minimum price for required items of a composite product, depending on the current selection of item options.
+	 * This would be used for a composite product that has required items without default selections.
+	 */
+	const selectCompositeProductMinPrice = createSelector(
+		selectProductEntities,
+		selectCompositeProductAppliedOptionsEntitiesState,
+		(products, appliedOptionsEntities, props) => {
+			const product = selectProduct.projector(products, { id: props.id });
+			if(product.type !== DaffProductTypeEnum.Composite) {
+				return undefined;
+			}
+			const appliedOptions = selectCompositeProductAppliedOptions.projector(appliedOptionsEntities, { id: props.id });
+
+			return (<DaffCompositeProduct>product).items.reduce((acc, item) => daffAdd(
+				acc, 
+				appliedOptions[item.id].value ? item.options.find(option => option.id === appliedOptions[item.id].value).price : getMinimumCompositeItemPrice(item)
```

Comment:
I think you should extract this into:

```ts
function findAppliedCompositeOptionPrice(item: DaffCompositeProductItem, appliedOptions: Dictionary<DaffCompositeProductEntityItem>): number {
  return item.options.find(option => option.id === appliedOptions[item.id].value).price
}
```

and use it in `selectCompositeProductPrice`, `selectCompositeProductMinPrice`, and `selectCompositeProductMaxPrice`.

Reply:
done